chainset.go

```diff
@@ -105,27 +120,17 @@ func (cs *chainSet) CreateCommonAccount(ctx context.Context, keyName string) (fa
 func (cs *chainSet) Start(ctx context.Context, testName string, additionalGenesisWallets map[ibc.Chain][]ibc.WalletData) error {
 	for c := range cs.chains {
 		c := c
-		if c.Config().Type == "rollapp" {
-			seq, err := c.CreateRollapp(testName, ctx, additionalGenesisWallets[c]...)
-			cs.seq = seq
-			if err != nil {
-				return fmt.Errorf("failed to start chain %s: %w", c.Config().Name, err)
-			}
-		}
-	}
-	for c := range cs.chains {
-		c := c
-		if c.Config().Type == "hub" {
-			if err := c.StartHub(testName, ctx, cs.seq, additionalGenesisWallets[c]...); err != nil {
+		if _, ok := c.(ibc.RollHub); ok {
+			if err := c.Start(testName, ctx, cs.seq, additionalGenesisWallets[c]...); err != nil {
 				return fmt.Errorf("failed to start chain %s: %w", c.Config().Name, err)
 			}
 		}
 	}
 
 	for c := range cs.chains {
 		c := c
-		if c.Config().Type == "rollapp" {
-			if err := c.StartRollapp(testName, ctx, additionalGenesisWallets[c]...); err != nil {
+		if _, ok := c.(ibc.RollHub); !ok {
+			if err := c.Start(testName, ctx, cs.seq, additionalGenesisWallets[c]...); err != nil {
```

Comment:
Do we always need seq for the start function?

Reply:
We need it because this is the base interface of chain

Reply:
Removed it